Bugsnag/include/Bugsnag/Bugsnag.h

```diff
@@ -301,16 +301,25 @@
  *  Add a callback to be invoked before a session is sent to Bugsnag.
  *
  *  @param block A block which can modify the session
+ *
+ *  @returns An opaque reference to the callback which can be passed to `removeOnSession:`
  */
-+ (void)addOnSessionBlock:(BugsnagOnSessionBlock _Nonnull)block
++ (nonnull BugsnagOnSessionRef)addOnSessionBlock:(nonnull BugsnagOnSessionBlock)block
     NS_SWIFT_NAME(addOnSession(block:));
 
 /**
  * Remove a callback that would be invoked before a session is sent to Bugsnag.
  *
- * @param block The block to be removed.
+ * @param callback The opaque reference of the callback to be removed.
```

Comment:
Maybe be explicit that this **must** be the value returned by `addOnSessionBlock`?

Reply:
Updated the comments 👍